api/files.py

```diff
@@ -34,9 +34,20 @@ def getHashingFieldStorage(upload_dir, hash_alg):
     class HashingFieldStorage(cgi.FieldStorage):
         bufsize = 2**20
         def make_file(self, binary=None):
+            if not self.filename:
+                return self.file
             self.open_file = HashingFile(os.path.join(upload_dir, os.path.basename(self.filename)), hash_alg)
             return self.open_file
 
+        # override private method __write of superclass FieldStorage
+        # _FieldStorage__file is the private variable __file of the same class
+        def _FieldStorage__write(self, line):
+            if self._FieldStorage__file is not None:
```

Comment:
Two (possibly dumb) questions:

1. Why override the _superclass_ method? Why not just create our own `__write` method? Wouldn't that have the same effect? Presumably it's called by `self.__write` which would work, yes?
1. Similarly, why wouldn't `self.__file` work? I don't think superclass members are isolated from subclasses, are they?